src/Command/Tunnel/TunnelOpenCommand.php

```diff
@@ -0,0 +1,199 @@
+        }
+
+        $logFile = $tunnelDir . '/log.txt';
+        $log = new StreamOutput(fopen($logFile, 'a'));
+
+        ProcessUtil::fork();
+
+        /** @var Process[] $processes */
+        $processes = [];
+        $error = false;
+        foreach ($relationships as $name => $services) {
+            foreach ($services as $key => $service) {
+                $remoteHost = $service['host'];
+                $remotePort = $service['port'];
+
+                $start = isset($localPort) ? $localPort + 1 : 30000;
+                $localPort = PortUtil::getPort($start);
+                $tunnelInfo = [
+                    'projectId' => $project->id,
+                    'environmentId' => $environment->id,
+                    'appName' => $appName,
+                    'relationship' => $name,
+                    'key' => $key,
+                    'remotePort' => $remotePort,
+                    'remoteHost' => $remoteHost,
+                    'localPort' => $localPort,
+                    'pid' => null,
+                ];
+                $pidFile = $this->getPidFile($tunnelInfo, $tunnelDir);
+
+                $processes[$pidFile] = $this->createTunnelProcess($sshUrl, $remoteHost, $remotePort, $localPort);
+
+                try {
+                    $processes[$pidFile]->start(function ($type, $buffer) use ($log) {
+                        $log->writeln($buffer);
+                    });
+                }
+                catch (\Exception $e) {
+                    $this->stdErr->writeln(sprintf('Failed to open tunnel: %s', $e->getMessage()));
+                    unset($processes[$pidFile]);
+                    $error = true;
+                    continue;
+                }
+
+                $tunnelInfo['pid'] = $processes[$pidFile]->getPid();
+                file_put_contents($pidFile, $tunnelInfo['pid']);
+
+                $this->stdErr->writeln(sprintf('SSH tunnel opened at %s:%s to %s.%d', 'localhost', $localPort, $name, $key));
+
+                // Wait a small time to capture any immediate errors.
+                usleep(100000);
+
+                if (!$processes[$pidFile]->isRunning() && !$processes[$pidFile]->isSuccessful()) {
+                    $this->stdErr->writeln(trim($processes[$pidFile]->getErrorOutput()));
+                    unlink($pidFile);
+                    unset($processes[$pidFile]);
+                    $error = true;
+                }
+            }
+        }
+
+        if (count($processes)) {
+            $this->stdErr->writeln("Logs are written to: $logFile");
+
+            // @todo
+            $this->stdErr->writeln("There's no way to close the tunnels yet... delete their PID files manually with:\n    rm $tunnelDir/*.pid");
+        }
+
+        ProcessUtil::killParent($error);
+
+        if (!count($processes)) {
+            return $error ? 1 : 0;
+        }
+
+        while (count($processes)) {
+            sleep(1);
+            foreach (array_keys($processes) as $pidFile) {
```

Comment:
Isn't it possible to have `$error === true` at this point? It seems to be completely ignored afterwards.